.github/actions/coverage-report/action.yml

```diff
@@ -0,0 +1,104 @@
+name: Coverage Report
+
+inputs:
+  GH_TOKEN:
+    required: true
+  BOOST_VERSION:
+    required: true
+  THRIFT_VERSION:
+    required: true
+  RUN_TESTS:
+    required: true
+  HAZELCAST_ENTERPRISE_KEY:
+    required: true
+  AWS_ACCESS_KEY_ID:
+    required: true
+  AWS_SECRET_ACCESS_KEY:
+    required: true
+  HZ_TEST_AWS_INSTANCE_PRIVATE_IP:
+    required: true
+
+env:
+  # Not possible to set this as a default
+  # https://github.com/orgs/community/discussions/46670
+  shell: bash
+
+runs:
+  using: composite
+  steps:
+    - name: Read Config
+      shell: ${{ env.shell }}
+      run: cat .github/config.env >> $GITHUB_ENV
+
+    - name: Install Necessary Packages
+      shell: ${{ env.shell }}
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y net-tools libssl-dev gdb gcovr lcov curl
+
+    - name: Download hazelcast-enterprise-tests.jar
+      shell: ${{ env.shell }}
+      run: |
+        curl -H "Authorization: token ${{ inputs.GH_TOKEN }}" https://raw.githubusercontent.com/hazelcast/private-test-artifacts/data/certs.jar > hazelcast-enterprise-${{ env.HAZELCAST_VERSION }}-tests.jar
+
+    - name: Install Boost
+      shell: ${{ env.shell }}
+      run: |
+        sudo ./scripts/install-boost.sh ${{ inputs.BOOST_VERSION }}
+
+    - name: Install Thrift
+      shell: ${{ env.shell }}
+      run: |
+        sudo ./scripts/install-thrift.sh ${{ inputs.THRIFT_VERSION }}
+
+    - name: Setup JDK
+      uses: actions/setup-java@v4
+      with:
+        java-version: ${{ env.JAVA_VERSION }}
+        distribution: ${{ env.JAVA_DISTRIBUTION }}
+
+    - name: Build
+      env:
+        BUILD_DIR: build
+        COVERAGE: ON
+      shell: ${{ env.shell }}
+      run: |
+        ./scripts/build-unix.sh                                          \
```

Comment:
Remove unwanted spaces for all below
`                                          \`

Reply:
e750c0a05471a4e094758590b315a2ec492ad6fc